app/addons/fauxton/components.react.jsx

```diff
@@ -0,0 +1,170 @@
+
+      if (!version) { return null; }
+      return (
+        <div className="version-footer">
+          Fauxton on 
+          <a href="http://couchdb.apache.org/">Apache CouchDB</a>
+          <br/> 
+          v. {version}
+        </div>
+      );
+    }
+  });
+
+  var Burger = React.createClass({
+    render: function () {
+      return (
+        <div className="burger" onClick={this.toggleMenu}>
+          <div></div>
+          <div></div>
+          <div></div>
+        </div>
+      );
+    },
+
+    toggleMenu: function () {
+      Actions.toggleNavbarMenu();
+    }
+  });
+
+  var NavLink = React.createClass({
+    render: function () {
+      var link = this.props.link;
+      var liClassName = this.props.active === link.title ? 'active' : '';
+
+      return (
+        <li data-nav-name={link.title} className={liClassName} >
+          <a href={link.href}>
+            <i className={link.icon + " fonticon "}></i>
+            {link.title}
+          </a>
+        </li>
+      );
+    }
+  });
+
+  var NavBar = React.createClass({
+    getStoreState: function () {
+      return {
+        navLinks: navBarStore.getNavLinks(),
+        bottomNavLinks: navBarStore.getBottomNavLinks(),
+        footerNavLinks: navBarStore.getFooterNavLinks(),
+        activeLink: navBarStore.getActiveLink()
+      };
+    },
+
+    getInitialState: function () {
+      return this.getStoreState();
+    },
+
+    createLinks: function (links) {
+      return _.map(links, function (link, i) {
+        return <NavLink key={i} link={link} active={this.state.activeLink} />;
+      }, this);
+    },
+
+    onChange: function () {
+      this.setState(this.getStoreState());
+    },
+
+    componentDidMount: function () {
+      navBarStore.on('change', this.onChange, this);
+    },
+
+    componentWillUnmount: function() {
+      navBarStore.off('change', this.onChange);
+    },
+
+    render: function () {
+      var navLinks = this.createLinks(this.state.navLinks);
+      var bottomNavLinks = this.createLinks(this.state.bottomNavLinks);
+      var footerNavLinks = this.createLinks(this.state.footerNavLinks);
+
+      return (
+        <div className="navbar">
+          <Burger />
+          <nav id="main_navigation">
+            <ul id="nav-links" className="nav">
+              {navLinks}
+            </ul>
+
+            <div id="bottom-nav">
+              <ul id="bottom-nav-links" className="nav">
+                {bottomNavLinks}
+              </ul>
+            </div>
+          </nav>
+
+          <div className="bottom-container">
+            <div className="brand">
+              <div className="icon">Apache Fauxton</div>
+            </div>
+            <Footer />
+            <div id="footer-links">
+              <ul id="footer-nav-links" className="nav">
+                {footerNavLinks}
+              </ul>
+            </div>
+          </div>
+        </div>
+      );
+    }
+  });
+
+
+  return {
```

Comment:
[Opinion Alert!] I *really* like this pattern of having a return statement at the end of a file that does little more than outline what's being returned, e.g.

```javascript
return {
   Burger: Burger,
   Fries: Fries,
   Coffee: Coffee
};
```

I find it much easier to read at a glance what's available without having to sift through code. We don't do this in Fauxton - we just append to an object as it goes down the file then return the object. Be nice to start doing it...